idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,411 @@
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2019, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Centrifugal compressor stage model.  This model is based on:
+
+Modekurti et al., (2017). "Design, Dynamic Modeling, and Control of a Multistage CO2
+				Compresor System." International Journal of Greenhouse Gas Control. v62.
+Created: April 2020
+__Author__ = "Quang Minh Le "
+"""
+
+from __future__ import division
+# import Pyomo
+from pyomo.environ import (ConcreteModel, SolverFactory, Reals, NonNegativeReals,
+    Var, Param, Expression, Constraint, SolverFactory, value, exp, log, sqrt)
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+
+# Import IDAES cores
+from idaes.core import (declare_process_block_class,
+                        FlowsheetBlock)
+from idaes.generic_models.unit_models.pressure_changer import (PressureChangerData,
+																ThermodynamicAssumption)
+
+from idaes.generic_models.properties.helmholtz.helmholtz import (
+    HelmholtzThermoExpressions as ThermoExpr
+)
+import idaes.logger as idaeslog
+from idaes.core.util import from_json, to_json, StoreSpec
+
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+import idaes
+import math
+
+
+_log = idaeslog.getLogger(__name__)
+
+@declare_process_block_class("InletStage", doc="Inlet stage of compressor model")
+class InletStageData(PressureChangerData):
+    # Pressure changer with isentropic compressor options
+    CONFIG = PressureChangerData.CONFIG()
+    CONFIG.compressor = True
+    CONFIG.get("compressor")._default = True
+    CONFIG.get("compressor")._domain = In([True])
+    CONFIG.thermodynamic_assumption = ThermodynamicAssumption.isentropic
+    CONFIG.get("thermodynamic_assumption")._default = \
+        ThermodynamicAssumption.isentropic
+    CONFIG.get('thermodynamic_assumption')._domain = \
+        In([ThermodynamicAssumption.isentropic])
+
+    ################################################
+
+    def build(self):
+        super(InletStageData, self).build()
+        config = self.config # shorter config pointer
+
+        # The thermodynamic expression writer object, te, writes expressions
+        # including external function calls to calculate thermodynamic quantities
+        # from a set of state variables.
+        te = ThermoExpr(blk=self, parameters=config.property_package)
+
+        if SolverFactory('ipopt').available():
```

Comment:
There should be no reason to have a solver in the `build` method, especially one which i dependent on a specific solver.

Reply:
Yes, I removed those lines in the revised version.